packages/components/src/RichTextEditor/RichTextEditor/RichTextEditor.tsx

```diff
@@ -89,8 +91,9 @@ export const RichTextEditor = ({
   const [schema] = useState<ProseMirrorModel.Schema>(
     createSchemaFromControls(controls)
   )
-  const [labelId] = useState<string>(labelledBy || reactId)
-  const [editorId] = useState<string>(reactId)
+
+  const [editorId] = useState<string>(id || reactId)
```

Comment:
While you're here, `reactId` is a bad var name, mind renaming to `editorId` or similar?

Reply:
Cant use editorId since its already the name of the state, I was contemplating renaming it `generatedId`

Reply:
You don't need the state, so you just can do `editorId` instead of `reactId` and remove this line.

Reply:
The `editorId` can either be the generated Id from useId() or the prop Id, so we still need both

Reply:
You can just do `const editorId = id || useId()` I believe, but this is fine